File: java/nanocontainer/src/java/org/nanocontainer/integrationkit/PicoCompositionException.java
package org.nanocontainer.integrationkit;

import org.picocontainer.PicoException;

public class PicoCompositionException extends PicoException {

    protected Throwable cause;

    protected PicoCompositionException() {
        super();
    }

    public PicoCompositionException(String message, Throwable cause) {
        super(message);
        this.cause = cause;
    }

    public PicoCompositionException(Throwable cause) {
        this.cause = cause;
    }

    public PicoCompositionException(String message) {
        super(message);
        cause = null;
    }

    public Throwable getCause() {
        return cause;
    }
}

File: trunk/dotnet/nanocontainer/src/IntegrationKit/PicoCompositionException.cs
using System;
using System.Runtime.Serialization;
using PicoContainer.Core;

namespace NanoContainer.IntegrationKit
{
	public class PicoCompositionException : PicoException
	{
		public PicoCompositionException()
		{
		}

		public PicoCompositionException(Exception ex) : base(ex)
		{
		}

		public PicoCompositionException(string message) : base(message)
		{
		}

		public PicoCompositionException(string message, Exception ex) : base(message, ex)
		{
		}

		protected PicoCompositionException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
		}
	}
}